weave/trace_server/current_calls_query.sql

```diff
@@ -0,0 +1,203 @@
+-- From the 100 limited calls we get their usage data
+-- From the llm ids in the usage data we get the prices and rank them
+-- We get the top ranked prices and discard the rest
+-- We join the top ranked prices with the usage data to get the token costs
+-- Finally we pull all the data from the calls and add a costs object
+
+-- First we get lightly filtered calls, to optimize later for heavy filters
+WITH 
+    filtered_calls AS (
+        SELECT calls_merged.id AS id
+        FROM calls_merged
+        WHERE project_id = {pb_2_1:String}
+        AND (id IN {pb_2_2:Array(String)})
+        GROUP BY (project_id,
+                id)
+        HAVING (((any(calls_merged.deleted_at) IS NULL))
+                AND (calls_merged.id IN {pb_2_0:Array(String)}))
+        ORDER BY any(calls_merged.started_at) ASC
+        LIMIT 1),
+
+                -- Then we get all the calls we want, with all the data we need, with heavy filtering
+                all_calls AS (
+        SELECT calls_merged.project_id AS project_id,
+            calls_merged.id AS id,
+            any(calls_merged.op_name) AS op_name,
+            argMaxMerge(calls_merged.display_name) AS display_name,
+            any(calls_merged.trace_id) AS trace_id,
+            any(calls_merged.parent_id) AS parent_id,
+            any(calls_merged.started_at) AS started_at,
+            any(calls_merged.ended_at) AS ended_at,
+            any(calls_merged.exception) AS
+        exception,
+            array_concat_agg(calls_merged.input_refs) AS input_refs,
+            array_concat_agg(calls_merged.output_refs) AS output_refs,
+            any(calls_merged.wb_user_id) AS wb_user_id,
+            any(calls_merged.wb_run_id) AS wb_run_id,
+            any(calls_merged.deleted_at) AS deleted_at,
+            any(calls_merged.attributes_dump) AS attributes_dump,
+            any(calls_merged.inputs_dump) AS inputs_dump,
+            any(calls_merged.output_dump) AS output_dump,
+            any(calls_merged.summary_dump) AS summary_dump
+        FROM calls_merged
+        WHERE project_id = {pb_2_3:String}
+        AND (id IN filtered_calls)
+        GROUP BY (project_id,
+                id)
+        ORDER BY any(calls_merged.started_at) ASC),
+
+    -- From the all_calls we get the usage data for LLMs
+    -- Generate a list of LLM IDs and their respective token counts from the JSON structure
+    llm_usage AS (
+        
+        SELECT
+            id,
+            started_at,
+            ifNull(JSONExtractRaw(summary_dump, 'usage'), '{}') AS usage_raw,
+            arrayJoin(
+                arrayMap(
+                    kv -> (kv.1, kv.2),
+                    JSONExtractKeysAndValuesRaw(usage_raw)
+                )
+            ) AS kv,
+            kv.1 AS llm_id,
+            JSONExtractInt(kv.2, 'requests') AS requests,
+            -- Some libraries return input_tokens and output_tokens, others prompt_tokens and completion_tokens
+            if(
```

Comment:
This is really interesting - another great chance to standardize this with a better type system. For backwards compat, we should keep both, but i think we might want to formally define the expectations at the interface layer and also conform to that in our python client